### PR TITLE
feat(smithy-build): support maven section for vscode smithy plugin

### PR DESCRIPTION
**Pull request by cogwirrel**
The VSCode Smithy Plugin requires maven dependencies and repositories to be specified in the smithy-build.json file, so we add these as optional properties to synthesize.

https://github.com/awslabs/smithy-vscode/blob/main/README.md#authoring-a-model

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/smithy/smithy-build.ts [hunk 1]
```diff
@@ -51,6 +69,11 @@ export interface SmithyBuildOptions extends SmithyCommon {
    * @default - no ignoreMissingPlugins set in the smithy-build.json file
    */
   readonly ignoreMissingPlugins?: boolean;
+  /**
+   * Maven configuration, used to declare dependencies for the smithy vs-code plugin
+   * @see https://github.com/awslabs/smithy-vscode/blob/main/README.md#authoring-a-model
+   */
+  readonly maven?: SmithyMavenConfig;
```
Comment (mrgrain) [inline src/smithy/smithy-build.ts:76]: This just needs an `@default` description and we are good to go.

**Review by mrgrain — CHANGES_REQUESTED** [review]
Looks good. Got a small doc request.

File: src/smithy/smithy-build.ts (hunk 1 above)
Comment (cogwirrel) [reply src/smithy/smithy-build.ts]: Thanks for the speedy review! :) Updated :)

**Review by mrgrain — APPROVED** [review]
🚀

**Pull request merged**